src/params.jl

```diff
@@ -10,6 +10,13 @@ Parameters for ParticleDA run. Keyword arguments:
 * `enable_timers::Bool` : Flag to control run time measurements
 * `particle_save_time_indices`: Set of time indices to save particles at
 * `seed`: Seed to initialise state of random number generator used for filtering
+* `n_tasks`: Number of tasks to use for running parallelisable operations. Positive
+   integers indicate the number of tasks directly, while the absolute value of negative
+   integers indicate the number of task to use per-thread (as reported by 
+   `Threads.nthreads()`). Using multiple tasks per thread will improve the ability of
+   the scheduler to balance load across threads but potentially increase overheads.
+   If simulation of the model being filtered use multiple threads then it may be 
+   beneficial to set the `n_tasks = 1` to avoid too much contention between threads.
```

Comment:
Not sure if this overly complex, but couldn't think of a nice way otherwise of allowing either setting total number of tasks explicitly (useful when we want to not use thread-based parallelism across particles as for example model itself is using multiple threads) or as a multiple of number of threads. Could have two separate parameters for these, for example `n_tasks` and `n_tasks_per_thread` but then unclear which takes precedence.

Reply:
@DanGiles just tagging to get your thoughts here